update Makefile go to v1.20.4 and linter to 1.52.2

jluhrsen opened the pull request:
- one whitespace error fixed as part of the lint upgrade

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/ovn-org/ovn-kubernetes/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Trivial changes are exempt from following this template.
If your change is non-trivial, please provide the following information:
-->

**- What this PR does and why is it needed**
<!--
A summary of the changes within this pull request and some context
as to why they were made
-->

**- Special notes for reviewers**
<!--
What exactly did you change - you may also defer to information
contained in commit messages. At a bare minimum it's worth highlighting
which areas of the code were changed as it's easier to assign reviewers
-->


**- How to verify it**
<!--
Did you include unit tests? or end-to-end tests?
How can I manually verify that this patch achieves its objective
-->


**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

dcbw reviewed (approved):
LGTM

dcbw commented:
Although... 1.20.4 is absolute latest go; can we double-check with @girishmg and his team if they are OK with this?

trozet commented:
in 4.14 we are using 1.19: rhel-9-golang-1.19-openshift-4.14

I think we should go with that.

jluhrsen commented:
I will update it to 1.19

jluhrsen commented:
> I will update it to 1.19

@trozet @girishmg , it's updated now.

tssurya commented:
lgtm for the 1.19 version, agree 1.20.4 is too early.
@jluhrsen : are these in preparation for 1.27 rebase PR?

tssurya commented on go-controller/Makefile:
```diff
@@ -11,7 +11,7 @@ PKGS ?=
 GOPATH ?= $(shell go env GOPATH)
 TEST_REPORT_DIR?=$(CURDIR)/_artifacts
 export TEST_REPORT_DIR
-GO_VERSION ?= 1.18.4
```
our go.mod is still using 1.18 shouldn't we update that to 1.19 as well?

jluhrsen replied:
I can... It's in the [1.27 rebase](https://github.com/ovn-org/ovn-kubernetes/pull/3593). I was just trying to get these changes in place so that rebase PR would
be able to build and test properly.

tssurya replied:
right I'd suggest a 1.27 rebase test with 1.19 and see if it works - if it does update go.mod updates to 1.19 in this PR itself? if not then it means we need the 1.20 for some dependency..

jluhrsen replied:
ok, testing locally in my 1.27 rebase branch was fine with 1.19. It's now updated here to be 1.19 as
well. Once this is in, I'll get back to the rebase PR.

flavio-fernandes replied:
@jluhrsen With this Makefile change, I am now getting the following issue when doing a `make test`:

```
vagrant@devmaster ~/dev/ovn-kubernetes.git/go-controller master*
❯ make test
podman run -it --rm --security-opt label=disable --cap-add=NET_ADMIN --cap-add=SYS_ADMIN -v /home/vagrant/dev/ovn-kubernetes.git:/go/src/github.com/ovn-org/ovn-kubernetes -w /go/src/github.com/ovn-org/ovn-kubernetes/go-controller -e COVERALLS= -e GINKGO_FOCUS="" quay.io/giantswarm/golang:1.19.6 sh -c "RACE=1 DOCKER_TEST=1 COVERALLS= PKGS="" hack/test-go.sh focus \"\" "
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
```

Any ideas how to work around that?

flavio-fernandes replied:
This works:
```
❯ podman run -it --rm --security-opt label=disable --cap-add=NET_ADMIN --cap-add=SYS_ADMIN -v /home/vagrant/dev/ovn-kubernetes.git:/go/src/github.com/ovn-org/ovn-kubernetes -w /go/src/github.com/ovn-org/ovn-kubernetes/go-controller -e COVERALLS= -e GINKGO_FOCUS="" quay.io/giantswarm/golang:1.18.4 sh -c "RACE=1 DOCKER_TEST=1 COVERALLS= PKGS="" hack/test-go.sh focus \"\" "
...
This works!!!
```

```
vagrant@devmaster ~/dev/ovn-kubernetes.git/go-controller master*
❯ podman run -it --rm --security-opt label=disable --cap-add=NET_ADMIN --cap-add=SYS_ADMIN -v /home/vagrant/dev/ovn-kubernetes.git:/go/src/github.com/ovn-org/ovn-kubernetes -w /go/src/github.com/ovn-org/ovn-kubernetes/go-controller -e COVERALLS= -e GINKGO_FOCUS="" quay.io/giantswarm/golang:1.19.6 sh -c "RACE=1 DOCKER_TEST=1 COVERALLS= PKGS="" hack/test-go.sh focus \"\" "
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
```

jluhrsen replied:
hopefully #3643 fixes this

jluhrsen commented:
> lgtm for the 1.19 version, agree 1.20.4 is too early. @jluhrsen : are these in preparation for 1.27 rebase PR? --> I get this feeling we need to move to 1.20 for the 1.27 release right? or is that not a requirement? in OCP I think we moved to 1.20 as part of rebase..?

Tim points out that we are still building with 1.19 for our 4.14 OCP branch. But, yes, I see that some OCP projects have
moved to 1.20 in their go.mod.

but, yes this is in preparation for the 1.27 rebase. PR is [here](https://github.com/ovn-org/ovn-kubernetes/pull/3593)

tssurya reviewed (approved):
lgtm

tssurya commented:
btw: what about https://github.com/ovn-org/ovn-kubernetes/blob/119b745002d12621a00cfe5de60e9075cb3a31e2/test/e2e/go.mod#L3 ? do we want to do this only via: https://github.com/ovn-org/ovn-kubernetes/pull/3516 ?

jluhrsen commented:
> btw: what about
> 
> https://github.com/ovn-org/ovn-kubernetes/blob/119b745002d12621a00cfe5de60e9075cb3a31e2/test/e2e/go.mod#L3
> 
> ? do we want to do this only via: #3516 ?

yes, @bpickard22 advised me to do it this way. I guess just bumping it in the test go.mod would create a lot of new problems being addressed in #3516

jluhrsen commented:
/retest
